### ISSUE TITLE
injected code results in errors that emacs hates

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#3530]
For example:

```
<core-macros>:3:10: 3:61 note: in expansion of #error
```

Emacs wants to load `<core-macros>` and interrupts your workflow to prompt for it. Injected code should come from files on disk (like intrinsic.rs) and we should use the correct file path when inserting them into the AST, so that the error messages work.

Alternately, we can come up with another way to indicate injected code so that the diagnostic emitter can generate error messages that emacs doesn't complain about. This could be a better solution, since getting dumped into core_macros.rs may not be so useful.

**Comment by steveklabnik** [comment]
Triage: I don't use emacs, but these errors are the same, so I'm guessing this is still the same.

**Comment by steveklabnik** [comment]
Moved to rust-lang/rust-mode#4
